cli/cliui/agent_test.go

```diff
@@ -672,3 +675,139 @@ func TestPeerDiagnostics(t *testing.T) {
 		})
 	}
 }
+
+func TestConnDiagnostics(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name  string
+		diags cliui.ConnDiags
+		want  []*regexp.Regexp
+	}{
+		{
+			name: "Direct",
+			diags: cliui.ConnDiags{
+				ConnInfo:     &workspacesdk.AgentConnectionInfo{},
+				PingP2P:      true,
+				LocalNetInfo: &tailcfg.NetInfo{},
+			},
+			want: []*regexp.Regexp{
+				regexp.MustCompile(`^✔ You are connected directly, peer-to-peer \(p2p\)$`),
+			},
+		},
+		{
+			name: "DirectBlocked",
+			diags: cliui.ConnDiags{
+				ConnInfo: &workspacesdk.AgentConnectionInfo{
+					DisableDirectConnections: true,
+				},
+			},
+			want: []*regexp.Regexp{
+				regexp.MustCompile(`^❗ You are connected via a DERP relay, not directly, peer-to-peer \(p2p\)$`),
+				regexp.MustCompile(`^❗ Your Coder administrator has blocked direct connections$`),
+			},
+		},
+		{
+			name: "NoStun",
+			diags: cliui.ConnDiags{
+				ConnInfo: &workspacesdk.AgentConnectionInfo{
+					DERPMap: &tailcfg.DERPMap{},
+				},
+				LocalNetInfo: &tailcfg.NetInfo{},
+			},
+			want: []*regexp.Regexp{
+				regexp.MustCompile(`^❗ You are connected via a DERP relay, not directly, peer-to-peer \(p2p\)$`),
```

Comment:
None of these use any regex related things so seems like it could just be `[]string` and `require.Contains` instead